src/theory/fp/fp_converter.cpp

```diff
@@ -820,868 +820,469 @@ FpConverter::uf FpConverter::buildComponents(TNode current)
-                 "expandDefinition";
-          break;
-
-        // Again, no action is needed
-        case kind::FLOATINGPOINT_COMPONENT_EXPONENT:
-        case kind::FLOATINGPOINT_COMPONENT_SIGNIFICAND:
-        case kind::ROUNDINGMODE_BITBLAST:
-        /* Fall through ... */
-
-        default: CVC5_FPCONV_PASSTHROUGH; break;
-      }
-    }
-    else if (t.isReal())
-    {
-      switch (current.getKind())
-      {
-        /******** Conversions ********/
-        case kind::FLOATINGPOINT_TO_REAL_TOTAL:
+        else if (kind == kind::FLOATINGPOINT_TO_SBV_TOTAL)
         {
-          // We need to recurse so that any variables that are only
-          // used under this will have components created
-          // (via auxiliary constraints)
-
-          TypeNode childType(current[0].getType());
-          fpMap::const_iterator arg1(d_fpMap.find(current[0]));
-
-          if (arg1 == d_fpMap.end())
-          {
-            workStack.push_back(current);
-            workStack.push_back(current[0]);
-            continue;  // i.e. recurse!
-          }
-
-          // However we don't need to do anything explicitly with
-          // this as it will be treated as an uninterpreted function
-          // by the real theory and we don't need to bit-blast the
-          // float expression unless we need to say something about
-          // its value.
+          Assert(d_rmMap.find(cur[0]) != d_rmMap.end());
+          Assert(d_fpMap.find(cur[1]) != d_fpMap.end());
+          FloatingPointToSBVTotal info =
+              cur.getOperator().getConst<FloatingPointToSBVTotal>();
+
+          d_sbvMap.insert(
+              cur,
+              symfpu::convertFloatToSBV<traits>(fpt(cur[1].getType()),
+                                                (*d_rmMap.find(cur[0])).second,
+                                                (*d_fpMap.find(cur[1])).second,
+                                                info.d_bv_size,
+                                                sbv(cur[2])));
         }
-
-        break;
-
-        case kind::FLOATINGPOINT_TO_REAL:
-          Unreachable()
-              << "Partially defined fp.to_real should have been removed by "
-                 "expandDefinition";
-          break;
-
-        default: CVC5_FPCONV_PASSTHROUGH; break;
+        // else do nothing
       }
     }
     else
     {
-      CVC5_FPCONV_PASSTHROUGH;
+      Assert(visited.at(cur) == 1);
+      continue;
     }
   }
 
-  return result;
+  if (d_boolMap.find(node) != d_boolMap.end())
+  {
+    Assert(node.getType().isBoolean());
+    return (*d_boolMap.find(node)).second;
+  }
+  if (d_sbvMap.find(node) != d_sbvMap.end())
+  {
+    Assert(node.getKind() == kind::FLOATINGPOINT_TO_SBV_TOTAL);
+    return (*d_sbvMap.find(node)).second;
+  }
+  if (d_ubvMap.find(node) != d_ubvMap.end())
+  {
+    Assert(node.getKind() == kind::FLOATINGPOINT_TO_UBV_TOTAL);
+    return (*d_ubvMap.find(node)).second;
+  }
+  return node;
```

Comment:
Why does `convert` return the original node for floating point and rounding mode nodes? @martin-cs

Reply:
Afaiu this is because via `registerTerm`, we get a couple of node kinds for which we do nothing. One example is `kind::FLOATINGPOINT_TO_REAL_TOTAL`. I can try to figure out the whole set, but we would skip them anyways...

Reply:
Ah wait, I think I misunderstood your question. You were asking why we only return a word-blasted node for boolean and `to_ubv` and `to_sbv` nodes. Yes, I was wondering about that, too. If we move the word-blasting to `preNotifyFact` which I'm currently evaluating, it might actually make sense to assert that the node is Boolean. Except that we still have to word-blast leafs.

Reply:
Ummm... will look at.  IIRC it is that it has to return something and there is nothing directly to do for these.

Reply:
Alternatively, we could also return a null node. And I also have an update on my previous comment, we can assert that the given node is Boolean when we move word-blasting to `preNotifyFact`.